VTK Composite Pipeline Bug Fix

NickGeneva opened the pull request:
After a lot of debugging, I believe this gets the job done.

This fix is two parts, first removal of Update from custom code wrapper to make sure inputs have been initialized before updating. Plus this was redundant. (Should probably centralize VTK object updating for safety purposes, but this works). This was discussed in Issue #34.

The second part of this fix is a small consistency check when requesting output ports of VTK objects. 
- Nodes such as the info node, ping the connected input node to draw the information text by blender (so even outside of pressing update).
- This calls [get_output](https://github.com/tkeskita/BVtkNodes/blob/master/core.py#L245) function of the connected input node requesting an output port is the vtk object exists on that node.
- It is my understanding from my debugging that **if the input connections are not initialized and output ports of the vtk object** are called this will throw the error in the composite pipeline.
- I _think_ this is due to VTK trying to build a vtkAlgorithm object, which gets confused because is wants an output but has nothing to input!

This simple check makes sure the number of initialized inputs of the vtkobject is consistent with the number of inputs in self.m_connections()[0] (defined for each node). This uses [GetTotalNumberOfInputConnections](https://vtk.org/doc/nightly/html/classvtkAlgorithm.html#aaccc64bf1e8d7392f7fb66b7744954b2), a vtk algorithm function, so I think this will exist for all vtk nodes??

tkeskita commented:
Hi,
thanks for your work on this! Your idea sounds correct, and seems to work otherwise but I found out that `vtkPlane` in the [cutting example](https://bvtknodes.readthedocs.io/en/latest/ug_nodes.html#cutting-field-data) crashes at this error, could you please look into this? Note: vtkPlane is a customized node.
```
AttributeError: 'vtkmodules.vtkCommonDataModel.vtkPlane' object has no attribute 'GetTotalNumberOfInputConnections'
```

NickGeneva commented:
Added check in get_ouput to make sure vtkobj is vtk.vtkAlgorithm for the return cases that require a vtk.Algorithm (ie. new check and also getoutputport). vtkPlane works now since its output is "self"

Also added a check in the info node for get number of points since some nodes do not contain this method.

Verified for all examples on my end.

tkeskita commented:
Seems to work for me too, thanks!
